cmds/vpdbootmanager/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/systemboot/systemboot/pkg/booter"
+	"github.com/systemboot/systemboot/pkg/vpd"
+)
+
+const usage = `Usage:
+boot add [netboot [dhcpv6|dhcpv4] [MAC] | localboot [grub|path [Device GUID] [Kernel Path]]]
+
+Ex.
+boot add localboot grub
+
+boot add netboot dhcpv6 AA:BB:CC:DD:EE:FF
+
+Use -h to show flag options.
+`
+
+func main() {
+	flag.Usage = func() {
+		fmt.Println(usage)
+	}
+	if err := parse(os.Args[1:]); err != nil {
+		flag.Usage()
+		fmt.Printf("Error: %s\n\n", err)
+		os.Exit(1)
```

Comment:
here and above we are mixing flag and os, we should not. The options are:
1) we use `flag` entirely, i.e. `flag.Parse()`, `flag.Args` and so on (with the additional benefit that you don't have to handle the parsing error yourself, it will detect it, print an error, and exit accordingly)
2) we don't use `flag` at all, we keep the parsing functions below, and just pass command arguments to the relevant functions.

If you go with 1), `flag` can handle positional arguments. Basically anything that comes after the first non-switch (anything that doesn't start with "-") argument, is captured and returned via `flag.Args()`, and you can pass `flag.Args()[1:]` to the subcommand handler.

If you go with 2), just keep your parsing functions and pass the arguments around, but no mixing with `flag`

Reply:
We cannot use `flag.Args` here (which I would have loved to use) because we have `FlagSet`s further down in the parsing line. As in, for `flag.Args` to work we need to do `flag.Parse()` which will stip the arguments needed to be kept for our `FlagSet`s. So I guess it's option 2 here, will amend with that.